src/client.js

```diff
@@ -3038,8 +3017,11 @@ MatrixClient.prototype.getTurnServers = function() {
  *
  * @param {Boolean=} opts.disablePresence True to perform syncing without automatically
  * updating presence.
+ * @param {Boolean=} opts.lazyLoadMembers True to not load all membership events during
+ * initial sync but fetch them when needed by calling `loadOutOfBandMembers`
+ * This will override the filter option at this moment.
  */
-MatrixClient.prototype.startClient = function(opts) {
+MatrixClient.prototype.startClient = async function(opts) {
```

Comment:
Hmm - in terms of backwards compat, what will happen if a client doesn't wait on this promise and starts calling API methods immediately?

Reply:
Good point. `MatrixClient::_syncApi` wouldn't be set yet, which could have some side-effects:
 - `getSyncState` returning null
 - `startClient` twice could leave you with two running `SyncApi`s
 - `stopClient` wouldn't stop the sync
 - `retryImmediately` would crash

Hmm, I guess this could be a breaking change then, and hence if we want to do it carry a major version number change.

Reply:
Maybe I just keep the filter creation inside MatrixClientPeg then?

Reply:
(out of band: let's do a version bump)